packages/dapp/components/lib/hooks/use-balances.tsx

```diff
@@ -26,24 +28,34 @@ export const BalancesContext = createContext<[Balances | null, RefreshBalances]>
 export const BalancesContextProvider: React.FC<ChildrenShim> = ({ children }) => {
   const [balances, setBalances] = useState<Balances | null>(null);
   const [walletAddress] = useWalletAddress();
-  const managedContracts = useManagerManaged();
   const namedContracts = useNamedContracts();
+  const protocolContracts = useProtocolContracts();
+  const { provider } = useWeb3();
 
   async function refreshBalances() {
-    if (walletAddress && managedContracts && namedContracts) {
+    if (!walletAddress || !namedContracts || !protocolContracts || !provider) {
+      return;
+    }
+
+    if(protocolContracts.managerFacet && protocolContracts.dollarToken && protocolContracts.creditToken && protocolContracts.governanceToken && protocolContracts.creditNft && protocolContracts.stakingShare) {
```

Comment:
this was acceptable on the other PR because is lesser code, but this pattern won't be acceptable on this one, find other way around to achieve this

Reply:
> this was acceptable on the other PR because is lesser code, but this pattern won't be acceptable on this one, find other way around to achieve this

Have you checked that file before the last commit?
```
  async function refreshBalances() {
    if (walletAddress && namedContracts && protocolContracts && provider) {
      const _3crvToken = protocolContracts.managerFacet && await protocolContracts.managerFacet.curve3PoolTokenAddress();
      const dollar3poolMarket = protocolContracts.managerFacet && await protocolContracts.managerFacet.stableSwapMetaPoolAddress();
      const _3crvTokenContract = getERC20Contract(_3crvToken, provider);
      const dollarMetapool = getIMetaPoolContract(dollar3poolMarket, provider);

      const [uad, _3crv, uad3crv, ucr, ubq, ucrNft, stakingShares, usdc, dai, usdt] = await Promise.all([
        protocolContracts.dollarToken && protocolContracts.dollarToken.balanceOf(walletAddress),
        _3crvTokenContract.balanceOf(walletAddress),
        dollarMetapool.balanceOf(walletAddress),
        protocolContracts.creditToken && protocolContracts.creditToken.balanceOf(walletAddress),
        protocolContracts.governanceToken && protocolContracts.governanceToken.balanceOf(walletAddress),
        erc1155BalanceOf(walletAddress, protocolContracts.creditNft),
        erc1155BalanceOf(walletAddress, protocolContracts.stakingShare),
        namedContracts.usdc.balanceOf(walletAddress),
        namedContracts.dai.balanceOf(walletAddress),
        namedContracts.usdt.balanceOf(walletAddress),
      ]);

      setBalances({
        uad,
        _3crv,
        uad3crv,
        ucr,
        ucrNft,
        ubq,
        stakingShares,
        usdc,
        dai,
        usdt,
      });
    }
  }
  ```
  It's also not acceptable?

Reply:
`protocolContracts` can't be null because its' already defined as an object.. could we possibly define the type of the receiving variable as any?
```
const protocolContracts: any = useProtocolContracts();
```
Whole codebase is as follows:
```
export const BalancesContextProvider: React.FC<ChildrenShim> = ({ children }) => {
  const [balances, setBalances] = useState<Balances | null>(null);
  const [walletAddress] = useWalletAddress();
  const namedContracts = useNamedContracts();
  const protocolContracts: any = useProtocolContracts();
  const { provider } = useWeb3();

  async function refreshBalances() {
    if (!walletAddress || !namedContracts || !protocolContracts || !provider) {
      return;
    }
   
    const _3crvToken = await protocolContracts.managerFacet.curve3PoolTokenAddress();
    const dollar3poolMarket = await protocolContracts.managerFacet.stableSwapMetaPoolAddress();
    const _3crvTokenContract = getERC20Contract(_3crvToken, provider);
    const dollarMetapool = getIMetaPoolContract(dollar3poolMarket, provider);

    const [uad, _3crv, uad3crv, ucr, ubq, ucrNft, stakingShares, usdc, dai, usdt] = await Promise.all([
      protocolContracts.dollarToken.balanceOf(walletAddress),
      _3crvTokenContract.balanceOf(walletAddress),
      dollarMetapool.balanceOf(walletAddress),
      protocolContracts.creditToken.balanceOf(walletAddress),
      protocolContracts.governanceToken.balanceOf(walletAddress),
      erc1155BalanceOf(walletAddress, protocolContracts.creditNft),
      erc1155BalanceOf(walletAddress, protocolContracts.stakingShare),
      namedContracts.usdc.balanceOf(walletAddress),
      namedContracts.dai.balanceOf(walletAddress),
      namedContracts.usdt.balanceOf(walletAddress),
    ]);

    setBalances({
      uad,
      _3crv,
      uad3crv,
      ucr,
      ucrNft,
      ubq,
      stakingShares,
      usdc,
      dai,
      usdt,
    });
  }

  useEffect(() => {
    refreshBalances();
  }, [walletAddress, protocolContracts]);

  return <BalancesContext.Provider value={[balances, refreshBalances]}>{children}</BalancesContext.Provider>;
};
```

Reply:
Found a solution.. funny thing

Reply:
we already know that `protocolContracts` object is not null, so `Non-null Assertion Operator` can be used, I think